microservices/gatewayJobScheduler/app.py

```diff
@@ -34,11 +34,33 @@ def transform_data_by_ns(data):
                         break
 
                 for host in route_obj['hosts']:
+                    # Look for a matching certificate by SNI for custom domains
```

Comment:
Can this new block of logic be moved to a separate function that returns `cert` and `cert_id`?  Also, not sure if the Kong `id` is a sufficient identifier for determining there is a changed cert, because a user can update the `cert.cert` and `cert.key` while keeping the kong `id` the same.  You could hash the `cert.cert` and use that, or inspect the cert itself to get the serial number.  Alternatively, don't bother passing the `sslCertificateId` and let kube api dig out the serial number for comparison.